gammapy/datasets/flux_points.py

```diff
@@ -122,11 +139,52 @@ def __init__(
         self.models = models
         self.meta_table = meta_table
 
-        if mask_safe is None:
-            mask_safe = (~data.is_ul).data
+        self.available_stat_type = dict(
```

Comment:
`FluxPointsDataset.available_stat_types` can be a property like in `FluxPoints.available_quantities`